### PR TITLE
Tooling maintenance

### PR DESCRIPTION
**Pull request by AlexeyTsvetkov**
Gradle plugin:

* Update Gradle to 8.3;
* Update Gradle plugins;
* Start using version catalogs;
* Update dependencies;

Intellij plugin:

* Update Gradle to 8.3;
* Enable Configuration Cache;
* Start using version catalogs;

### CODE REVIEW

File: gradle-plugins/compose/build.gradle.kts [hunk 1]
```diff
@@ -77,7 +75,9 @@ val shadow = tasks.named<ShadowJar>("shadowJar") {
     val fromPackage = "de.undercouch"
     val toPackage = "org.jetbrains.compose.$fromPackage"
     relocate(fromPackage, toPackage)
-    archiveClassifier.set("shadow")
+    archiveBaseName.set("shadow")
+    archiveClassifier.set("")
```
Comment (dima-avdeev-jb) [inline gradle-plugins/compose/build.gradle.kts:79]: archiveClassifier changes from "shadow" to empty string ""
Is it correct?

File: gradle-plugins/gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[versions]
+kotlin = "1.9.0"
+gradle-download-plugin = "5.5.0"
+kotlinx-serialization = "1.2.1"
```
Comment (dima-avdeev-jb) [inline gradle-plugins/gradle/libs.versions.toml:4]: latest kotlinx-serialization version is 1.6.0
Can we consider upgrade? Or better to keep as it was been before?

**Review by dima-avdeev-jb — APPROVED** [review]

File: gradle-plugins/gradle/libs.versions.toml (hunk 2 above)
Comment (dima-avdeev-jb) [reply gradle-plugins/gradle/libs.versions.toml:4]: Oh, I saw another PR. (https://github.com/JetBrains/compose-multiplatform/pull/3607)
That's fine to keep 1.2.1 for now

File: gradle-plugins/compose/build.gradle.kts (hunk 1 above)
Comment (AlexeyTsvetkov) [reply gradle-plugins/compose/build.gradle.kts:79]: Yes, for some reason, the new version requires to remove the classifier

**Pull request merged**